lbrynet/daemon/Daemon.py

```diff
@@ -1841,8 +1842,25 @@ def jsonrpc_publish(self, name, bid, metadata=None, file_path=None, fee=None, ti
             }
         }
 
+        claim_copy = deepcopy(claim_dict)
         if sources is not None:
             claim_dict['stream']['source'] = sources
+            claim_copy['stream']['source'] = sources
+        elif file_path is not None:
+            if not os.path.isfile(file_path):
+                raise Exception("invalid file path to publish")
+            claim_copy['stream']['source'] = {
+                'version': '_0_0_1',
+                'sourceType': 'lbry_sd_hash',
+                'source': '0' * 96,
+                'contentType': ''
+            }
+        else:
+            raise Exception("no source provided to publish")
```

Comment:
The documentation says "If no path is given but a metadata dict is the source from the given metadata will be used". This doesn't seem to be intended behavior here. Maybe update the doc, or change the logic?

Reply:
updated the docstring